10a-walkthrough-6.Rmd

```diff
@@ -0,0 +1,677 @@
+
+```{r}
+all_files %>% map(names)
+```
+
+That means that we can combine all six datasets into one using `bind_rows`. We'll call this newly combined dataset `child_counts`:
+
+```{r}
+child_counts <-  all_files %>% 
+    bind_rows()
+```
+
+Since we know that a) each of our six datasets had eight variables and b) our combined dataset also has eight variables, we can conclude that all our rows combined together correctly. But let's use `str` to verify: 
+
+```{r}
+str(child_counts)
+```
+
+## Filter for the desired disabilities and age groups
+
+We want to explore gender related variables, but our dataset has additional aggregate data for other subgroups. For example, we can use `count` to explore all the different disability groups in the dataset: 
+
+```{r}
+child_counts %>% 
+    count(`SEA Disability Category`)
+```
+
+Since we will be visualizing and modeling gender variables for all students in the dataset, we'll filter out all subgoups except "All Disabilities" and the age totals: 
+
+```{r}
+child_counts <- child_counts %>% 
+    filter(`SEA Disability Category` == "All Disabilities", 
+           `SEA Education Environment` %in% c("Total, Age 3-5", "Total, Age 6-21")) 
+```
+
+## Rename the variables
+
+In the next section we'll be preparing the dataset for visualization and modeling by "tidying" it. When we write code to transform datasets, we have to repeatedly type the column names so it's useful to change them to ones with reasonable lengths and without spaces. 
+
+```{r}
+child_counts <- child_counts %>% 
+    rename(year = Year,
+           state = "State Name", 
+           age = "SEA Education Environment",
+           disability = "SEA Disability Category", 
+           f_3_5 = "Female Age 3 to 5", 
+           m_3_5 = "Male Age 3 to 5", 
+           f_6_21 = "Female Age 6 to 21", 
+           m_6_21 = "Male Age 6 to 21")
+```
+
+## Clean state names
+
+You might have noticed that some state names in our dataset are in upper case letters and some are in lower case letters: 
+
+```{r}
+child_counts %>% 
+    count(state) %>% head()
+```
+
+If we leave it like this, R will treat the upper case states and lower case states of the same name as different states. We can use `mutate` and `tolower` to transform all the state names to lowercase letters. 
+
+```{r}
+child_counts <- child_counts %>% 
+    mutate(state = tolower(state)) 
+```
+
+## Tidy the dataset 
+
+Visualizing and modeling our data will be much easier if our dataset is in a "tidy" format. In his paper *Tidy Data*, Hadley Wickham defines tidy datasets where
```

Comment:
cite? also, should we use a standard citation format (should we use a .bib file?)